doc/source/whatsnew/v0.21.0.txt

```diff
@@ -233,6 +233,30 @@ Dtype Conversions
 - Inconsistent behavior in ``.where()`` with datetimelikes which would raise rather than coerce to ``object`` (:issue:`16402`)
 - Bug in assignment against ``int64`` data with ``np.ndarray`` with ``float64`` dtype may keep ``int64`` dtype (:issue:`14001`)
 
+Dropping argmin/argmax support for arbitrary dtypes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:func:`Series.argmin` and :func:`Series.argmax` will now raise a ``TypeError`` when used with ``object`` dtypes. Previously, these functions would attempt to coerce their arguments to floats, which usually led to a ``ValueError``. Now, ``object`` dtypes are explicitly disallowed with ``argmax`` and related functions. For example, the old behavior produced a somewhat confusing message:
+
+.. code-block:: ipython
+
+  In [1]: s = pd.Series(['foo', 'bar'])
+
+  In [2]: s.argmax()
+  ...
+  ValueError: could not convert string to float: 'bar'
+
+This has now changed to
+
+.. code-block:: ipython
+
+  In [1]: s = pd.Series(['foo', 'bar'])
```

Comment:
Sounds good. Right, I think what happened is:

- before the PR, `argmin` would raise `ValueError` with strings and objects
- while working on the PR, the fix in the `nanops` module allowed these functions to work with strings and objects, but only if there were no missing values. I updated some test cases to reflect this behavior.
- Our discussion led to deciding that this behavior is too inconsistent, that fixing it in general would be pretty complicated, so the best thing for now is to explicitly disallow `nanargmin` with `object` dtypes.
- so now, `argmin` raises `TypeError` with strings and objects.